src/kerberos.cc

```diff
@@ -37,6 +37,13 @@ std::string ToStringWithNonStringAsEmpty(Napi::Value value) {
     return value.As<String>();
 }
 
+int BooleanToIntWithNonIntAsFalse(Napi::Value value) {
+    if (!value.IsBoolean()) {
+        throw Error::New(value.Env(), "Expected a boolean value");
```

Comment:
```suggestion
        throw TypeError::New(value.Env(), "Expected a boolean value");
```

1. I think TypeError is the correct error to throw here
2. Could we somehow incorporate the key name in the error message?  An error of `"expected a boolean value"` isn't actionable for users.  Instead, could we throw an error like `"options.protect to be a boolean or undefined"`.